src/bot.ts

```diff
@@ -302,6 +303,7 @@ export const getBot = async (bot: Bot<GrammyContext>) => {
 
   // Main message composer
   notChannelComposer.use(messagesComposer);
+  notChannelComposer.use(denylistComposer);
```

Comment:
It would be best if you used the `denylistComposer` composer inside the messages composer. Please look at the examples above of the `noAntisemitismComposer` composer.